Mini ch 1

cylcrow opened the pull request:
#### What this PR do?
OMG 😬 
Shows cards based on mocked API data. 😅 

#### Any background context you want to provide?
No, I guess... 🤔 

#### Where should the reviewer start?
See it [live](https://60309bdc258d66da90220bc5--blissful-goldwasser-8acdb6.netlify.app)

#### How should this be manually tested?
yarn test

#### Screenshots
![image](https://user-images.githubusercontent.com/78768286/108584837-083cc680-730a-11eb-8ea3-38c3ab20b30e.png)

cylcrow committed:
:white_check_mark: :heavy_plus_sign: :art: :fire: Fork, cleanup and first tests and component's creation

- Removed unnecessary files
- Added first tests
- Added first components
- Added dependencies for testing

cylcrow committed:
:construction: :white_check_mark: :art: Added tests to assert components default styling and stubs for not yet created components

cylcrow committed:
:recycle: Refactored names to comply with the specified in the challenge description

cylcrow committed:
:recycle: :white_check_mark: :sparkles: :lipstick: Added base components located in Header component

- First iteration for base components
- Modified Header structure: created StyledSection component to a better arrangement of components

cylcrow committed:
:sparkles: :recycle: :white_check_mark: :lipstick: Added Card component

cylcrow committed:
:sparkles: :recycle: :white_check_mark: :lipstick: Added Card component

- Added card component
- Added HomeVideos component
- Added tests for new added components
- Added/refactored styles for HomeVideo and Card component
- Refactored tests for Card component

cylcrow committed:
:lipstick: :art: Polished styles for Card component

cylcrow committed:
testing-libraries-cleanup: :recycle: Removed unnecesary dependencies after tests refactor to comply just with react-testing-library

jparciga commented on src/components/Base/Card/Card.jsx:
```diff
@@ -0,0 +1,55 @@
+import styled from 'styled-components';
+
+const StyledDiv = styled.div`
+  overflow: hidden;
+  height: 345px;
+  width: 345px;
+  margin: 8px;
+  box-shadow: 2px 2px 2px 2px #ccc;
+  border-radius: 5px;
+
+  &:hover {
+    background: lightgray;
+  }
+`;
+
+const StyledCardContent = styled.div`
+  text-align: justify;
+  padding: 4px;
+  margin: 4px;
+`;
+
+const StyledCardImage = styled.img`
+  height: 140px;
+  width: 100%;
+  display: block;
+  background-size: cover;
+  background-repeat: no-repeat;
+  background-position: center;
+  object-fit: cover;
+`;
+
+const StyledCardTitle = styled.h2`
+  font-weight: normal;
+  font-size: 1.25rem;
+`;
+
+const StyledCardDescription = styled.p`
+  font-weight: lighter;
+  color: gray;
+`;
```
It might be a good idea to separate all this code into a new file to keep your code cleaner.

jparciga commented on src/components/Base/Card/Card.test.jsx:
```diff
@@ -0,0 +1,45 @@
+  return { container };
+}
+
+describe('Card', () => {
+  it('renders', () => {
+    const container = build();
+    expect(container).toMatchSnapshot();
+  });
+});
+
+describe('Card styles and props', () => {
+  it('applies default styling', () => {
+    const { firstChild } = build().container;
+    expect(firstChild).toHaveStyle('overflow: hidden');
+    expect(firstChild).toHaveStyle('height: 345px');
+    expect(firstChild).toHaveStyle('width: 345px');
+    expect(firstChild).toHaveStyle('margin: 8px');
+    expect(firstChild).toHaveStyle('box-shadow: 2px 2px 2px 2px #ccc');
+    expect(firstChild).toHaveStyle('border-radius: 5px');
+  });
+
+  it('has all passed props', () => {
+    const EXPECTED_IMAGE = 'https://i.picsum.photos/id/237/536/354.jpg';
+    const EXPECTED_TITLE = 'title';
+    const EXPECTED_DESCRIPTION = 'description';
+    const { container } = build(
+      <Card
+        image={EXPECTED_IMAGE}
+        title={EXPECTED_TITLE}
+        description={EXPECTED_DESCRIPTION}
+      />
+    );
+    const img = container.querySelector("img");
+    expect(img).toHaveProperty("src", EXPECTED_IMAGE);
+    expect(container.textContent).toContain(EXPECTED_TITLE);
+    expect(container.textContent).toContain(EXPECTED_DESCRIPTION);
+  });
+});
```
Excellent!

jparciga commented on src/components/HomeVideos/HomeVideos.test.jsx:
```diff
@@ -0,0 +1,36 @@
+});
+
+describe('HomeVideos shows home videos', () => {
+  it('displays all videos', () => {
+    const { container } = build();
+    const videos = getAllByTestId(container, (id) => id.includes('video-card-'));
+    expect( videos ).toHaveLength(data.items.length);
+  });
```
That's a good test case

jparciga reviewed (approved):
# Mini-Challenge 1 Feedback

Good job 👏 !

## Acceptance Criteria
✅ The header is rendered correctly.
✅ A list of videos obtained from a mock file is displayed in the Home View.
✅ CSS styles are applied correctly using the styled-components approach.

## Bonus Points
✅ No warnings or errors are logged in the browser console.
✅ The UI is responsive

# Mini Challenge 2 Feedback:

It's great that you have applied TDD 😃. Please include a test coverage report.

## Acceptance Criteria
✅ Meaningful test cases were implemented for components and sub-routines logic.
✅ All the test cases were successful.

<img width="304" alt="Screen Shot 2021-02-25 at 16 34 11" src="https://user-images.githubusercontent.com/3632628/109236235-d194ff00-7794-11eb-814d-b027ea1ac533.png">

## Bonus Points
❓Test coverage is above 60%. -> Test coverage report is missing.

cylcrow committed:
:bento: :wheelchair: :recycle: Added roles to each element

- Added roles to elements to improve accesibility
- Refactored tests to find elements by role instead of their id
